doc/developer/design/20230225_compute_introspection_schema.md

```diff
@@ -0,0 +1,212 @@
+
+`raw` variants, on the other hand, measure some time value in
+nanoseconds, whereas their corresponding non-`raw` views convert the
+times to the SQL interval type.
+
+Confusingly, not all times are reported as intervals; for example,
+`mz_scheduling_parks` has `slept_for` and `requested` columns, whose
+type is `bigint` and whose unit is undocumented (in fact, it is nanoseconds).
+
+#### Inconsistent presence of `sum` column in histograms
+
+Some histograms include a per-bucket sum, which allows consumers to
+compute average or total values precisely. Confusingly, not all
+histograms report such a column (e.g. `mz_worker_compute_delays` does
+not, whereas `mz_worker_raw_compute_delays` does).
+
+#### Lack of clarity around per-replica vs. global relations
+
+Many of the relations are logging sources that run locally in
+clusters, are not persisted, and report different data depending on
+the currently selected replica. Views based on these sources are also
+per-replica. It is not straightforward to tell from our docs which is
+the case for each relation.
+
+Furthermore, sometimes the same logical value is reported by two
+relations, one reporting `clusterd`'s view of the world, and the other
+reporting `environmentd`'s. For example, `mz_compute_frontiers`
+vs. `mz_cluster_replica_frontiers`. However, the names do not make it
+clear what the distinction is. 
+
+#### Redundant `mz_` in names.
+
+Every relation's name begins with `mz_`. This is redundant, because
+anyone querying these relations probably already understand that they
+are using Materialize. Even if not, they are already in a namespace
+called `mz_internal`.
+
+### Suggested fixes for schema inconsistencies
+
+* All per-worker relations should have a corresponding "global" view
+  where the results are aggregated across workers in the way most
+  likely to be useful (for example, summing, taking the minimum, or
+  restricting to worker 0). Because these global views are more likely
+  to be useful than the underlying sources, they should have the basic
+  name (e.g., `mz_arrangement_sharing`). The underlying source should
+  have a name with `per_worker` inserted (e.g.,
+  `mz_per_worker_arrangement_sharing`). Furthermore, the per-worker
+  variant should be clearly less prominent in documentation (either at
+  the bottom of the page, or in a "see more" section that has to be
+  clicked to expand).
+* The expectations about per-worker relations should be clearly
+  documented (e.g., we should clearly say whether the values for each
+  worker are expected to be the same during normal operation).
+* All relations that report a simple count or sum should be views based on
+  an `_internal` variant that maintains the count or sum in the
+  Differential diff field. We may consider renaming this, since it
+  conflicts confusingly with the overall name `mz_internal`.
+* `raw` variants should be removed, and the underlying dataflows
+  should be in terms of intervals.
```

Comment:
@frankmcsherry was 👎🏽 on this in the past because the underlying data has nanosecond precision and our interval type only has microsecond precision.

Reply:
I think having two copies of every relation involving times is way more noise than it's worth. We can either have nanos everywhere as @teskje suggests, or we can give up nanosecond precision in the underlying sources. (How often is sub-microsecond precision useful, anyway?)

Reply:
I discussed this with @umanwizard offline and we agree that keeping just the nanoseconds is fine. Users can still convert those to intervals themselves if they are fine with the loss in precision.

Based on brief testing, it seems like all duration values we currently produce are above 1000 ns. That said, there is no reason why this always has to be the case in the future. One might hope, for example, that our operators one day are fast enough to have sub-microsecond invocation times. So keeping the nanosecond-precision values is future-proof.